src/playercards/customizable/UpgradeSheetLibrary.ttslua

```diff
@@ -195,27 +217,72 @@ function updateCheckboxes(rowIndex)
   end
   local checkboxIndex = rowCheckboxFirstIndex[rowIndex]
   for col = 1, checkboxCount do
-    local pos = getCheckboxPosition(rowIndex, col)
     if col <= selected then
-      pos.y = Y_VISIBLE
+      local identifier = tostring(rowIndex) .. " " .. tostring(col)
+      if isLoading then
+        Wait.condition(
+          function()
+            Wait.frames(
+              function()
+                isLoading = false;
+                self.UI.setAttribute(identifier, "color", "#000000")
+              end,
+              1
+            )
+          end,
+          function()
+            return not self.UI.loading
+          end
+        )
+      else
+        self.UI.setAttribute(identifier, "color", "#000000")
+      end
     else
-      pos.y = Y_INVISIBLE
+      local identifier = tostring(rowIndex) .. " " .. tostring(col)
+      if isLoading then
+        Wait.condition(
+          function()
+            Wait.frames(
+              function()
+                isLoading = false;
+                self.UI.setAttribute(identifier, "color", "#48b02800")
+              end,
+              1
+            )
+          end,
+          function()
+            return not self.UI.loading
+          end
+        )
+      else
+        self.UI.setAttribute(identifier, "color", "#48b02800")
+      end
     end
-    self.editButton({
-      index = checkboxIndex,
-      position = pos
-    })
     checkboxIndex = checkboxIndex + 1
   end
 end
 
 function updateTextField(rowIndex)
   local inputIndex = rowInputIndex[rowIndex]
   if selectedUpgrades[rowIndex] ~= nil and selectedUpgrades[rowIndex].text ~= nil then
-    self.editInput({
-      index = inputIndex,
-      value = " " .. selectedUpgrades[rowIndex].text
-    })
+    if isLoading then
+      Wait.condition(
+        function()
+          Wait.frames(
+            function()
+              isLoading = false;
+              self.UI.setAttribute(rowIndex, "text", " " .. selectedUpgrades[rowIndex].text)
```

Comment:
What's the reason for the additional space at the front?

Reply:
the previous code also edited the text to have this space at the front, I wasn't sure if it was crucial and forgot it there

Reply:
Oh wow - I think it was just for display reason. I think we can safely remove that